packages/shared_preferences/shared_preferences/lib/shared_preferences.dart

```diff
@@ -14,12 +14,26 @@ import 'package:shared_preferences_platform_interface/shared_preferences_platfor
 class SharedPreferences {
   SharedPreferences._(this._preferenceCache);
 
-  static const String _prefix = 'flutter.';
+  static String _prefix = 'flutter.';
+  static bool _prefixHasBeenSet = false;
```

Comment:
I considered having a `defaultPrefix` and a nullable `prefix` and checking to see if prefix were null instead. This seemed "more readable" so I went with this.